### DISCUSSION TITLE
How to correctly fix the camera parameters and save an image of the scene?

### DISCUSSION THREAD

**Discussion by nimafo** [isl-org/Open3D#5932] [Q&A]
Hi all, 
I have a code in which the geometry of a box changes in each iteration within known XYZ bounds.
From a fixed, viewpoint and using the same camera, I want to capture a snapshot in each iteration to observe the behavior of my model after each run. 
but looking at the results, it seems that the settings are not kept the same in different iterations:

geometry size in the below three pictures, respectively:
- 3.156122777 x 5.048651987 x 5.27256857
- 11.93332875 x 3.52520350 x  3.937217469
- 11.96147930 x 4.34533669 x  11.48175857

![20230218_18_13_20](https://user-images.githubusercontent.com/45439717/219879642-51c3e5da-34f3-468c-96ec-6ee2087a6217.jpg)
![20230218_18_13_38](https://user-images.githubusercontent.com/45439717/219879644-36aa7f66-6abe-46ac-a006-c1a360bea1a8.jpg)
![20230218_18_14_02](https://user-images.githubusercontent.com/45439717/219879645-86c2d517-3ba6-4a75-88a3-84ece23cc5c7.jpg)

this is the visualization part of my program: 
```
# Create a visualization window and add the objects to it
vis = o3d.visualization.Visualizer()
vis.create_window(visible=False)
vis.add_geometry(mesh)
vis.add_geometry(pcd)
vis.add_geometry(pcd1)
vis.add_geometry(pcd2)
vis.add_geometry(wireframe_mesh)
for i in range(len(lines)):
    vis.add_geometry(lines[i])
# Set the camera extrinsic parameters
vis.get_view_control().set_zoom(6)
vis.get_view_control().set_lookat([4, 5, 3])
vis.get_view_control().set_front([-0.5, -0.5, +0.5])
vis.get_view_control().set_up([0.0, 0.0, 1.0])
vis.get_view_control().set_constant_z_far(300)

# Set the camera intrinsic parameters
cam_params = vis.get_view_control().convert_to_pinhole_camera_parameters()
cam_params.extrinsic = np.identity(4)   # No extrinsic transformation
cam_params.intrinsic.set_intrinsics(640, 480, 525, 525, 320, 240)  # Set the camera intrinsics
vis.get_view_control().convert_from_pinhole_camera_parameters(cam_params)

# Hide the visualization window and capture a screenshot of the window
vis.poll_events()
vis.update_renderer()
vis.capture_screen_image(f'{images_folder}{iterationName}.jpg')

# Close the visualization window
vis.destroy_window()
```

**Comment by cwreynolds** [comment]
This question is 19 months old, so you have probably found the answer or no longer care. But in case someone else comes here, like I did while searching for something else:

By default `vis.add_geometry(mesh)` recomputes the bounding box of scene geometry, and changes the view to show it all. You can suppress that with a second arg of `False`. See the [add_geometry  doc](https://www.open3d.org/docs/latest/python_api/open3d.visualization.Visualizer.html#open3d.visualization.Visualizer.add_geometry).
